### PR TITLE
Fix git method & add more tests

### PR DESCRIPTION
**Pull request by RexWzh**
## Main Changes

- [ ] Fix regex expression in `check_git_version`
- [ ] Fix URL format in `LeanGitRepo.commit_url` | `os.path.join` gives invalid URL with `\\` rather than `/` in windows
- [ ] Use `gitpython` in `clone_and_checkout`
- [ ] Add tests for `LeanGitRepo`

Overall, the changes are compatible with the original code and fix some of the errors on Windows.

## Pytest

One test failed due to a network problem.
image

Re-run the test.
image

### CODE REVIEW

File: src/lean_dojo/data_extraction/lean.py [hunk 1]
```diff
@@ -455,7 +457,7 @@ def is_lean4(self) -> bool:
 
     @property
     def commit_url(self) -> str:
-        return os.path.join(self.url, f"tree/{self.commit}")
+        return f"{self.url}/tree/{self.commit}"
```
Comment (yangky11) [inline src/lean_dojo/data_extraction/lean.py:460]: What if `self.url` ends with `/`?

**Comment by yangky11** [comment]
Running tests

File: src/lean_dojo/data_extraction/lean.py (hunk 1 above)
Comment (RexWzh) [reply src/lean_dojo/data_extraction/lean.py:460]: good question. It is striped here:
```py
object.__setattr__(self, "url", normalize_url(self.url))
```

where
```py
def normalize_url(url: str) -> str:
    return _URL_REGEX.fullmatch(url)["url"]  # Remove trailing `/`.
```

**Comment by yangky11** [comment]
Thanks!

Will merge into `main` after the other PR is ready to merge

**Pull request merged**